### DISCUSSION TITLE
How can the community help improve the approval process for PRs?

### DISCUSSION THREAD

**Discussion by cesarcoatl** [Homebrew/discussions#3020] [Everyday usage]
I have been contributing to Homebrew for a little over a year, but only recently I have seen that some PRs go without approval for days, especially on the `cask-versions` repo; which is where I contribute the most.

I am aware that having humans go through each PR is not an easy task, and something that I appreciate from maintainers like @miccal, @bevanjkay, @Rylan12, @cho-m, et. al.

Also I am aware that at least the Casks I tend to update (`azure-data-studio-insiders` and `visual-studio-code-insiders`) auto-update, and if a user installs a version that's not current, it will update automatically. So no harm here.

So back to my question.

How can the community help improve the approval process for PRs?

Thanks!

**Comment by neutric** [comment]
Making it pain free, that is if the maintainers just have to merge, tends to speed things up. So contribute by guiding other contributors on their way to a merge-ready PR.

**Reply by cesarcoatl** [reply]
Let's say the PR was created through `brew bump-cask-pr` and all checks pass, potentially qualifying it as a "merge-ready" PR.

What would it make it "pain free"?

**Reply by bevanjkay** [reply]
@thecesrom I believe what @neutric is referring to is the fact that there have been a number of PRs that are not "merge-ready" across the homebrew-cask-* repos that have been taking up some additional time for maintainers. Helping other users to get their PRs "merge-ready" helps the entire process along.

**Reply by cesarcoatl** [reply]
Thanks for clarifying that @bevanjkay.

In my particular case I created three PRs for `azure-data-studio-insiders`:

https://github.com/Homebrew/homebrew-cask-versions/pull/13392 (merged by @bevanjkay 🎉 )
https://github.com/Homebrew/homebrew-cask-versions/pull/13390 (closed and opened https://github.com/Homebrew/homebrew-cask-versions/pull/13392)
https://github.com/Homebrew/homebrew-cask-versions/pull/13382 (closed and opened https://github.com/Homebrew/homebrew-cask-versions/pull/13390)

In some cases some PRs, even though are _merge-ready_, have been labeled stale.

So how I can I help?

**Reply by neutric** [reply]
@thecesrom Apart from increasing maintainer availability by becoming a maintainer, you can immediately improve the situation for everybody by putting less strain on the infrastructure. Not sure what the policy is for `homebrew-cask-versions`, but in `homebrew-cask` you'd be asked to dial down the frequency, especially if the casks in question auto-update.

Also, in the five months that I have been actively contribution again, these past handful(s) of days have been the only time where things progress a little slower. Just to put things into perspective.

**Reply by cesarcoatl** [reply]
Checking [Homebrew/homebrew-cask-versions//CONTRIBUTING.md#nightly-builds-nightlies](https://github.com/Homebrew/homebrew-cask-versions/blob/master/CONTRIBUTING.md#nightly-builds-nightlies), there's the following recommendation:

>use of a [url do block](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#using-a-block-to-defer-code-execution) in conjunction with `version :latest` and `sha256 :no_check` is encouraged

Although I was under the impression that both were discouraged.

I can say that `azure-data-studio-insiders` and `visual-studio-code-insiders` do have a Monday through Friday release schedule, which might qualify them as "nightly" or "daily". But using `version :latest` and `sha256 :no_check` will require the removal of `livecheck` and `auto_updates`, which might come with its own set of problems as `brew upgrade` will try to upgrade `azure-data-studio-insiders` even though it auto-updates.

Thoughts?

**Reply by bevanjkay** [reply]
@thecesrom `brew upgrade` would only upgrade the casks if the `--greedy` flag is passed. In the case of applications that are updated nightly, and have an auto-update mechanism, using `version :latest` would certainly reduce the maintenance overhead.

**Reply by cesarcoatl** [reply]
https://github.com/Homebrew/homebrew-cask-versions/pull/13398 for `azure-data-studio-insiders`
